spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -223,8 +233,34 @@ else if (isCountingQuery && this.tree.isDelete()) {
 		}
 	}
 
+	private Supplier<Builder<?>> getQueryBuilderSupplier(Object[] parameters) {
```

Comment:
Maybe add a javadoc comment to help describe what this helper method is doing? I guess it's looking at what fields will be needed from the return obj and then constructing a query object containing only those fields?